crates/ruff/src/rules/pycodestyle/rules/too_many_blank_lines.rs

```diff
@@ -0,0 +1,92 @@
+use ruff_python_ast::source_code::Locator;
+use ruff_text_size::{TextRange, TextSize};
+
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::newlines::Line;
+
+/// ## What it does
+/// Checks for too many (>=3) blank lines.
+///
+/// ## Why is this bad?
+/// PEP 8 recommends the using blank lines as following:
+/// - Two blank lines are expected between functions and classes
+/// - One blank line is expected between methods of a class.
+///
+/// ## Example
+/// ```python
+/// def func1():
+///     pass
+///
+///
+///
+/// def func2():
+///     pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def func1():
+///     pass
+///
+///
+/// def func2():
+///     pass
+/// ```
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+#[violation]
+pub struct TooManyBlankLines(pub usize);
+
+impl AlwaysAutofixableViolation for TooManyBlankLines {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooManyBlankLines(nb_blank_lines) = self;
+        format!("Too many blank lines ({nb_blank_lines})")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Remove extraneous blank line(s)".to_string()
+    }
+}
+
+/// E303
+pub(crate) fn too_many_blank_lines(line: &Line, locator: &Locator) -> Option<Diagnostic> {
+    // Only check for too many blank lines starting from the first blank line of a (potential) series
+    // of blank lines (to avoid duplicate errors).
+    // Also ignore blank lines at the beginning of the file.
+    if line.start().to_u32() > 0
```

Comment:
Nit: You can compare with `TextSize::new()`
```suggestion
    if line.start() > TextSize::new(0)
```

This heuristic to determine if it is the first line fails if the file starts with an UTF BOM header. I recommend you pass a boolean indicating whether this is the first line or not.

Reply:
> Nit: You can compare with TextSize::new()

I've noticed that I can also do `0.into()`, I had no idea which one is best practice. I'll switch to the `TextSize::new()` way.